modules/microshift-data-cleaning-container-images.adoc

```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * microshift_troubleshooting/microshift-cleanup-data.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-data-cleaning-container-images_{context}"]
+= Cleaning all data and keeping the container images
+
+You can retain the {microshift-short} container images while cleaning all data by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` arguments.
+
+Keeping the container images helps speed up {microshift-short} restart after data clean up because the necessary container images are already present locally when you start the service.
+
+When you run the script with the `--all` and `--keep-images` arguments, you perform the following clean up actions:
+
+* Stop and disable all {microshift-short} services
+* Delete all {microshift-short} pods
+* Reset network configuration
+* Delete the `/var/lib/microshift` data directory
+* Delete OVN-K networking configuration
+
+[WARNING]
+====
+This option deletes all {microshift-short} data and user workloads. Use with caution.
+====
+
+.Prerequisites
+* You are logged into {microshift-short} as an administrator with root-user access.
+* You have filed a support case.
+
+.Procedure
+
+. Clean up all data and user workloads while retaining the {microshift-short} container images by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` argument, by entering the following command:
++
+[source,terminal]
+----
+$ sudo microshift-cleanup-data --all --keep-images
+----
++
+.Example output
+[source,terminal]
+----
+DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?
+1) Yes
+2) No
+#? Yes
+Stopping MicroShift services
+Disabling MicroShift services
+Removing MicroShift pods
+Deleting the br-int interface
+Killing conmon, pause and OVN processes
+Removing MicroShift configuration
+Removing OVN configuration
+MicroShift service was stopped
+MicroShift service was disabled
+Cleanup succeeded
+
+$ sudo crictl images | awk '{print $1}'
+IMAGE
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+registry.redhat.io/lvms4/topolvm-rhel9
+registry.redhat.io/openshift4/ose-csi-external-provisioner
+registry.redhat.io/openshift4/ose-csi-external-resizer
+registry.redhat.io/openshift4/ose-csi-livenessprobe
+registry.redhat.io/openshift4/ose-csi-node-driver-registrar
+registry.redhat.io/ubi9
+----
++
+[IMPORTANT]
+====
+The {microshift-short} service is stopped and disabled after you run the script.
+====
```

Comment:
Based on the SME's comment [here](https://github.com/openshift/openshift-docs/pull/80106#discussion_r1732778861), the second part of this output actually should be split into its own step, like this:
```suggestion
----
+
[IMPORTANT]
====
The {microshift-short} service is stopped and disabled after you run the script.
====

. Verify that the container images are still present by running the following command:
+
[source,terminal]
----
$ sudo crictl images | awk '{print $1}'
----
+
.Example output
[source,terminal]
----
IMAGE
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
registry.redhat.io/lvms4/topolvm-rhel9
registry.redhat.io/openshift4/ose-csi-external-provisioner
registry.redhat.io/openshift4/ose-csi-external-resizer
registry.redhat.io/openshift4/ose-csi-livenessprobe
registry.redhat.io/openshift4/ose-csi-node-driver-registrar
registry.redhat.io/ubi9
----

```